src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -16,15 +16,25 @@ namespace System.Text.Json
     internal struct WriteStack
     {
         /// <summary>
-        /// The number of stack frames when the continuation started.
+        /// Exposes the stackframe that is currently active.
         /// </summary>
-        private int _continuationCount;
+        public WriteStackFrame Current;
+
+        /// <summary>
+        /// Buffer containing all frames in the stack. For performance it is only populated for serialization depths > 1.
```

Comment:
`For performance it is only populated for serialization depths > 1.`

It appears `EnsurePushCapacity()` always initialize the array?

Reply:
Correct, but that only happens for `_count > 1`. This is in line with current behavior.